adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala

```diff
@@ -285,6 +285,77 @@ class AlignmentRecordRDDFunctions(rdd: RDD[AlignmentRecord])
     }).sortByKey().map(p => p._2)
   }
 
+  def adamAddMDTags(referenceFile: String, fragmentLengthOpt: Option[Int] = None): RDD[AlignmentRecord] =
+    adamAddMDTags(
+      fragmentLengthOpt match {
+        case Some(fragmentLength) =>
+          sc.loadSequence(referenceFile, fragmentLength = fragmentLength)
+        case _ =>
+          sc.loadSequence(referenceFile)
+      }
+    )
+
+  def adamAddMDTags(referenceFragments: RDD[NucleotideContigFragment]): RDD[AlignmentRecord] = {
+    val collectedRefMap =
```

Comment:
As future cleanup, we might want to make this object (the collectedRefMap) extend [ReferenceFile](https://github.com/bigdatagenomics/adam/blob/master/adam-core/src/main/scala/org/bdgenomics/adam/util/ReferenceFile.scala), and then change `getRefSeq` to `ReferenceFile.extract`. Then we could use this with both 2Bit files and etc.